ADAL/src/ADAuthenticationContext+Internal.m

```diff
@@ -109,7 +109,19 @@ + (ADAuthenticationError*)errorFromDictionary:(NSDictionary*)dictionary
         NSUUID* correlationId = [dictionary objectForKey:OAUTH2_CORRELATION_ID_RESPONSE] ?
                                 [[NSUUID alloc] initWithUUIDString:[dictionary objectForKey:OAUTH2_CORRELATION_ID_RESPONSE]]:
                                 nil;
-        return [ADAuthenticationError OAuthServerError:serverOAuth2Error description:errorDetails code:errorCode correlationId:correlationId];
+
+        ADErrorCode code = errorCode;
+        NSString* suberror = [dictionary objectForKey:@"suberror"];
+        if (suberror && [suberror isEqualToString:@"protection_policy_required"])
+        {
+            code = AD_ERROR_SERVER_PROTECTION_POLICY_REQUIRED;
+        }
+        
+        return [ADAuthenticationError OAuthServerError:serverOAuth2Error
+                                           description:errorDetails
+                                                  code:code
+                                         correlationId:correlationId
+                                              userInfo:suberror ? @{@"suberror":suberror} : nil];
```

Comment:
please add "suberror" key to public NSError header, so developers know how to look for it